src/http/headers.cr

```diff
@@ -9,13 +9,27 @@ struct HTTP::Headers
   record Key, name : String do
     forward_missing_to @name
 
-    def hash
-      h = 0
-      name.each_byte do |c|
-        c = normalize_byte(c)
-        h = 31 * h + c
+    def hash(hasher)
+      # hash 4 bytes at a row
+      c = 0
+      bytesize = name.bytesize
+      slice = name.to_slice
+      hasher.raw(bytesize)
+      while c + 4 < bytesize
```

Comment:
Yes, a slice is basically a pointer with a size (so accesses are bounded). No more, no less. Slices can allocate HEAP memory with `Slice(T).new(size)` or bounded to view any memory (HEAP or stack) with `Slice(T).new(pointer, size)`, which is the case here.